src/fhirtypes/ElementDefinition.ts

```diff
@@ -2924,9 +2943,17 @@ export class ElementDefinition {
     // We don't want to clone the reference to the StructureDefinition, so temporarily save it and remove it
     const savedStructDef = this.structDef;
     this.structDef = null;
+    // We don't want to clone the tree references either
+    const savedTreeParent = this.treeParent;
+    this.treeParent = undefined;
+    const savedTreeChildren = this.treeChildren;
+    this.treeChildren = undefined;
```

Comment:
This is something of a concession to the tests that compare two `ElementDefinition` instances using `expect.toEqual`. A newly created `ElementDefinition` will have `undefined at these two properties, so in order for a clone to match, it must also have `undefined`, not `null`. The other option is for us to implement a custom jest matcher that treats `null` and `undefined` as equivalent. That might be the best idea here, since it does look a little strange.

Reply:
Update: I tried implementing a custom matcher, but it basically means reimplementing lodash equality, so I'm just going to leave these as undefined.

Reply:
Yeah I think that is the right choice. It is a little weird to have the code be written to make the tests happy, but not weird enough to say we should reimplement lodash equality 😳